src/Groups/GAPGroups.jl

```diff
@@ -121,26 +121,29 @@ Return the exponent of `G`, i.e. the smallest positive integer `e` such that `g`
 """
 Base.exponent(x::GAPGroup) = GAP.Globals.Exponent(x.X)
 
+function _gap_rand(f, G::GAPGroup)
+    s = f(G.X)
+    return group_element(G, s)
+end
+
 """
     rand(G::Group)
 
 Return a random element of the group `G`.
 """
-function Base.rand(x::GAPGroup)
-   s = GAP.Globals.Random(x.X)
-   return group_element(x, s)
+function Base.rand(rng::Random.AbstractRNG, rs::Random.SamplerTrivial{Gr}) where Gr<:Oscar.GAPGroup
+   G = rs[]
+   return _gap_rand(GAP.Globals.Random, G)
 end
 
 """
     rand_pseudo(G::Group)
 
-Return a random element of the group `G`. It works faster than `rand`, but the elements are not necessarily equally distributed.
-"""
-function rand_pseudo(G::GAPGroup)
-   s = GAP.Globals.PseudoRandom(G.X)
-   return group_element(G,s)
-end
+Return a random element of the group `G`.
 
+`rand_pseudo` usually is faster than `rand`, however there is no guaranee on uniform distribution of the returned element.
```

Comment:
It's not just about speed, though; the group G might be a gigantic matrix group whose size we don't even know; so we can't just write down all elements in order and choose uniformly.

So we try our best to choose "as uniformly random as possible", and it works quite well in practice and converges to a uniform distribution rather quickly for most groups etc. etc. .

(Of course this is already an issue with the existing text).

```suggestion
Return a pseudo random element of the group `G`.

It is sometimes necessary to work with finite groups that we cannot effectively enumerate, e.g. matrix groups over finite fields. We may not even know the size of these groups. Yet many algorithms need to sample elements from the group "as randomly as possible", whatever that means; but also they need this *fast*.

The function `rand_pseudo` returns elements that are cheap to compute and somehow random, but makes no guarantees about their distribution.

For finitely presented groups, it returns random words of bounded length.

For finite permutation and matrix groups, it uses a variant of the product replacement algorithm.
For most inputs, the resulting stream of elements relatively quickly converges to a uniform distribution.
```